compiler/noirc_evaluator/src/ssa/opt/mod.rs

```diff
@@ -21,3 +22,19 @@ mod resolve_is_unconstrained;
 mod runtime_separation;
 mod simplify_cfg;
 mod unrolling;
+
+#[cfg(test)]
+pub(crate) fn assert_ssa_equals(mut ssa: super::Ssa, expected: &str) {
+    use crate::trim_leading_whitespace_from_lines;
+
+    ssa.normalize_ids();
+
+    let ssa = ssa.to_string();
+    let ssa = trim_leading_whitespace_from_lines(&ssa);
+    let expected = trim_leading_whitespace_from_lines(expected);
+
+    if ssa != expected {
+        println!("Expected:\n~~~\n{}\n~~~\nGot:\n~~~\n{}\n~~~", expected, ssa);
+        similar_asserts::assert_eq!(expected, ssa);
+    }
```

Comment:
I'm now trying to migrate [this test](https://github.com/noir-lang/noir/blob/82046be7790bb7128f481ace94b8a6841f76c88d/compiler/noirc_evaluator/src/ssa/opt/unrolling.rs#L516) to use your parser. It turns out that both input and output SSA in the comments are invalid. I figured out how to fix the input based on the errors that `Ssa::from_str(src).unwrap();` printed, but then got here and got confused by the output of `assert_ssa_equals(ssa, expected);`:

```text
Differences (-left|+right):
-fn main f0 {
-b0():
-    constrain Field 0
-    constrain Field 0
-    constrain Field 0
-    constrain Field 0
-    jmp b23()
-b23():
-    constrain Field 0
-    constrain Field 0
-    constrain Field 0
-    constrain Field 0
-    jmp b27()
-b27():
-    constrain Field 0
-    constrain Field 0
-    constrain Field 0
-    constrain Field 0
-    jmp b31()
-b31():
+acir(inline) fn main f0 {
+  b0():
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
+    jmp b1()
+  b1():
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
+    jmp b2()
+  b2():
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
+    constrain u1 0 == Field 1
     jmp b3()
-b3():
+  b3():
+    jmp b4()
+  b4():
     return Field 0
 }
```

There are two problems:
1. `similar_asserts::assert_eq!(expected, ssa);` uses the arguments in the opposite order as `assert_ssa_equals(mut ssa: super::Ssa, expected: &str)` receives them, so it's a bit confusing that _left_ in the output corresponds to _right_ in the input. FWIW I would use `assert_eq!(ssa, expected)`. 
2. I was thrown off by seeing expected `constrain Field 0` when I knew that `constrain v5` did not parse, it expected `==` to follow. I didn't understand why it's the one we expect in the end, how did it simplify my input `constrain v5 == Field 1` into this? Then I realised that it's coming from an unparsed and invalid string that was in the test.

On the 2nd point think it would be helpful if, in case there is a mismatch in the strings, to try to parse the expectation and print a warning that it's not even valid. It's okay to print both values because the parsed and processed output might help understand what it should be (assuming it's somewhat correct), but just so I stop treating it as gospel.

Reply:
Good ones! I just pushed fixed for both.

Reply:
Cool, a panic is good too, I guess we can print the SSA or procure an example from somewhere to figure out what it should be 🙏